chef/cookbooks/nova_dashboard/recipes/server.rb

```diff
@@ -17,6 +17,28 @@
 include_recipe "apache2::mod_wsgi"
 include_recipe "apache2::mod_rewrite"
 
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+# Multidomain can only be enabled if the API version of Keystone is
+# equal or bigger that v3.
+keystone_api_version = keystone_settings['api_version']
+keystone_internal_auth_url = keystone_settings['internal_auth_url']
+
+# TODO(aplanas): Remove this line when keystone API 'v3' works for
+# every OpenStack configuration file.
+if node["nova_dashboard"]["multidomain"] && keystone_api_version.to_f < 3
+  keystone_api_version = '3'
+  keystone_internal_auth_url = keystone_internal_auth_url.sub(%r{/v#{keystone_settings['api_version']}/$},
+                                                              "/v#{keystone_api_version}/")
```

Comment:
Uhm I tried. The method to search the keystone node is private and because this code will be dropped in a near? future, I do not want to mangle with a helper from a different library.

If you think that this is still a better approach, I will change the code, but as @rhafer says, looks like that `KeystoneHelper` was not designed to be used outside the `barclamp-keystone` (apart from the hash, of course).